content/en/docs/operations/talos/installation/oci.md

```diff
@@ -0,0 +1,258 @@
+    launch_create_volume_details {
+      display_name       = "cozy-dev1-node${count.index + 1}-data"
+      compartment_id     = var.compartment_id
+      size_in_gbs        = "512"
+      volume_creation_type = "ATTRIBUTES"
+      vpus_per_gb        = "10"
+    }
+    type = "paravirtualized"
+  }
+
+  shape_config {
+    memory_in_gbs = "32"
+    ocpus         = "4"
+  }
+}
+
+resource "oci_core_vnic_attachment" "cozy_dev1_vlan_vnic" {
+  count       = length(var.node_private_ips)
+  instance_id = oci_core_instance.cozy_dev1_nodes[count.index].id
+
+  create_vnic_details {
+    vlan_id = oci_core_vlan.cozy_dev1_vlan.id
+  }
+}
+```
+
+Apply the configuration using Terraform:
+
+```bash
+oci session authenticate --region us-ashburn-1 --profile-name=DEFAULT
+terraform init
+terraform apply
+```
+
+You'll get three machines with the following IPs:
+- `192.168.1.11`
+- `192.168.1.12`
+- `192.168.1.13`
+
+
+They also have a VLAN interface with the following subnet:
+- `192.168.100.0/24`
+
+This subnet will be used for the internal cluster communication.
+
+## Talos Configuration
+
+Use Talm to apply config and install Talos Linux on the drive.
+
+1. [Download](https://github.com/cozystack/talm/releases/latest) latest Talm binary and save it to `/usr/local/bin/talm`
+2. Make it executable:
+   ```
+   chmod +x /usr/local/bin/talm
+   ```
+
+### Installation with Talm
+
+1. Create directory for new cluster:
+   ```bash
+   mkdir -p mycluster
+   cd mycluster
+   ```
+
+2. Run the following command to initialize Talm for Cozystack:
+
+   ```bash
+   talm init -p cozystack
+   ```
+
+   After initializing, generate a configuration template with the command:
+
+   ```bash
+   talm -n 1.2.3.4 -e 1.2.3.4 template -t templates/controlplane.yaml -i > nodes/nodeN.yaml
+   ```
+
+3. Edit the node configuration file as needed.
+
+   - Update `hostname` to the desired name.
+     ```yaml
+     machine:
+       network:
+         hostname: node1
+     ```
+
+   - Add private interface configuration, and move `vip` to this section. This section isn’t generated automatically:
+     - `interface` - Obtained from the "Discovered interfaces" by matching options for the private interface.
+     - `addresses` - Use the address specified for Layer 2 (L2).
+
+     Example:
+     ```yaml
+     machine:
+       network:
+         interfaces:
+           - interface: eth0
+             addresses:
+               - 1.2.3.4/29
+             routes:
+               - network: 0.0.0.0/0
+                 gateway: 1.2.3.1
+           - interface: eth1
+             addresses:
+               - 192.168.100.11/24
+             vip:
+               ip: 192.168.100.10
+     ```
+
+**Execution steps:**
+
+1. Run `talm apply -f nodeN.yml` for all nodes to apply the configurations. The nodes will be rebooted and Talos will be installed on the disk.
+2. Execute bootstrap command for the first node in the cluster, example:
+   ```bash
+   talm bootstrap -f nodes/node1.yml
```

Comment:
This command doesn't work because of SSL errors and bootstrap has no `-i` flag.

Reply:
I realized that probably I need to manually add the machine's public IP in the `nodeN.yaml` under `machine.certSANs`, but at this point the machine would no longer accept `talm apply -f nodeN.yaml -i` and required a certificate to be provided. So here we go again with `terraform destroy && terraform apply`. It would be nice to have warnings that "You should absolutely specify the public IP here or you will have to start all over again" and/or guidance on how to get out of a seemingly unsolvable situation without nuking everything and starting from scratch.

Reply:
resolved